engine/platform/win32/sys_win.c

```diff
@@ -46,6 +62,92 @@ qboolean Sys_DebuggerPresent( void )
 	return IsDebuggerPresent();
 }
 
+static void Sys_SetTimerHighResolution( void )
+{	
+	HMODULE hndl;
+	pfnNtSetTimerResolution_t pNtSetTimerResolution;
+	pfnNtQueryTimerResolution_t pNtQueryTimerResolution;
+	
+	hndl = GetModuleHandle( "ntdll.dll" );
+	pNtSetTimerResolution = (pfnNtSetTimerResolution_t)GetProcAddress( hndl, "NtSetTimerResolution" );
+	pNtQueryTimerResolution = (pfnNtQueryTimerResolution_t)GetProcAddress( hndl, "NtQueryTimerResolution" );
+
+	if( pNtSetTimerResolution && pNtQueryTimerResolution )
+	{
+		// undocumented NT API functions, for some mystery reason works better than timeBeginPeriod
+		ULONG min, max, cur;
+		pNtQueryTimerResolution(&min, &max, &cur);
+		pNtSetTimerResolution(max, 1, &cur);
+	}
+	else
+	{
+		// more conventional method to set timer resolution, using as fallback
+		TIMECAPS tc;
+		timeGetDevCaps( &tc, sizeof( TIMECAPS ));
+		timeBeginPeriod( tc.wPeriodMin );
+	}
+}
+
+static DWORD WINAPI Sys_TimerThread( LPVOID a )
+{
+	double oldtime = 0;
+	LARGE_INTEGER delay;
+	while (true)
+	{
+		WaitForSingleObject( g_timer.hMutex, INFINITE ); // lock mutex
+		double realtime = Platform_DoubleTime();
+		double delta = g_timer.flInterval - ( realtime - oldtime );
+		oldtime = realtime;
+
+		if (delta > 0)
+			delta = 0;
+
+		delay.QuadPart = -1 * 1e7 * ( g_timer.flInterval + delta );
+		ReleaseMutex( g_timer.hMutex ); // unlock mutex
+
+		SetWaitableTimer( g_timer.hTimer, &delay, 0, NULL, NULL, 0 );
+		WaitForSingleObject( g_timer.hTimer, INFINITE );
+		SetEvent( g_timer.hEvent );
+	}
+}
+
+void Platform_Delay( double time )
+{
+	WaitForSingleObject( g_timer.hMutex, INFINITE ); // lock mutex
+	g_timer.flInterval = time;
+	ReleaseMutex( g_timer.hMutex ); // unlock mutex
+	WaitForSingleObject( g_timer.hEvent, INFINITE ); // wait when event will be triggered
+}
+
+void Platform_TimerInit( void )
+{
+	Sys_SetTimerHighResolution();
+	g_timer.hTimer = CreateWaitableTimer(NULL, TRUE, NULL);
+	g_timer.hEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+	g_timer.flInterval = 1.0 / 60.0;
```

Comment:
Same as for `Platform_TimerInit` for posix threads about dedicated servers.

Reply:
Also, maybe we need to target maxfps value here? Instead of hardcoded 60 Hz?

Reply:
> Also, maybe we need to target maxfps value here? Instead of hardcoded 60 Hz?

Don't think it's better than hardcode, in this case